chart/values.yaml

```diff
@@ -265,6 +265,8 @@ serviceAccount:
 
 podAnnotations: {}
 
+jobAnnotations: {}
```

Comment:
Sorry for the detail, but figured I'd go through my thinking for anyone who might drop in on this.

The original use case for this PR was around running Mastodon with Istio to provide zero trust security capabilities. This requires having an Envoy sidecar in pods to capture and manage traffic between pods (the mTLS aspects of zero trust). So for most pods, I need Istio sidecars enabled.

For job pods, the job ends but doesn't notify Envoy that it completed, so Envoy continues running. Since there's a process running in the job pod, Kubernetes considers the job to still be running. There are at least two ways to approach this: modify the entry point in the container to notify Envoy of the job completion so that Envoy exits (keeping the security aspects of having Envoy for jobs), or not load the Envoy sidecar at all and just run the job as it would run without Istio (maybe not as secure but also not directly public-facing). The second method seems the easiest for now.

If we go the annotation route, then either I turn off Istio sidecar loading for the mastodon namespace and add an annotation to non-job pods to load the sidecar, or I keep Istio sidecar loading on for the mastodon namespace and add an annotation to job pods to not load the sidecar. Either way, the annotations for job and non-job pods are different.

By adding support for a generic `jobAnnotations`, this PR is no longer about Istio, but it supports anyone wanting to run Mastodon with Istio. It also supports other cases where someone might need different annotations for job pods than non-job pods.

Reply:
isn't this just working around a bug in Envoy that could present itself in non-job situations as well? e.g. if a main web or Sidekiq pod crashes, would Envoy cause Kubernetes to still think that that container was running, leading to a new / different behavior than without Envoy? I'm not against the solution we've landed on here (treating one-off job containers differently from non-job contains seems straightforward, although I would wonder if there's a more idiomatic solution, since I'm not familiar with the details of Kubernetes), but the underlying usecase still feels like a bit of a bad kludge for the fact that the Envoy sidecar is misbehaving.

Reply:
I believe the problem is that the pod has multiple containers and Kubernetes will consider the pod to be failed if a container stops due to an error (thus cleaning up all of the other containers), but since a successful job doesn't exit with an error, Kubernetes doesn't clean up the other containers in the pod.

More details are available in https://medium.com/redbox-techblog/handling-istio-sidecars-in-kubernetes-jobs-c392661c4af7. It's been this way for a few years now.

Reply:
I added some in-line documentation. Ended up including `podAnnotations` in it to point out the differences.